packages/mobile/src/fiatExchanges/ProviderOptionsScreen.tsx

```diff
@@ -0,0 +1,160 @@
+import colors from '@celo/react-components/styles/colors'
+import fontStyles from '@celo/react-components/styles/fonts'
+import variables from '@celo/react-components/styles/variables'
+import { RouteProp } from '@react-navigation/native'
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useLayoutEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import {
+  Image,
+  SafeAreaView,
+  ScrollView,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+} from 'react-native'
+import BackButton from 'src/components/BackButton'
+import Dialog from 'src/components/Dialog'
+import { openMoonpay, openSimplex } from 'src/fiatExchanges/utils'
+import { CURRENCY_ENUM } from 'src/geth/consts'
+import i18n, { Namespaces } from 'src/i18n'
+import QuestionIcon from 'src/icons/QuestionIcon'
+import { moonpayLogo, simplexLogo } from 'src/images/Images'
+import { LocalCurrencyCode } from 'src/localCurrency/consts'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
+import { emptyHeader } from 'src/navigator/Headers'
+import { Screens } from 'src/navigator/Screens'
+import { TopBarIconButton } from 'src/navigator/TopBarButton'
+import { StackParamList } from 'src/navigator/types'
+import useSelector from 'src/redux/useSelector'
+import { useCountryFeatures } from 'src/utils/countryFeatures'
+import { currentAccountSelector } from 'src/web3/selectors'
+
+type Props = StackScreenProps<StackParamList, Screens.ProviderOptionsScreen>
+
+ProviderOptionsScreen.navigationOptions = ({
+  route,
+}: {
+  route: RouteProp<StackParamList, Screens.ProviderOptionsScreen>
+}) => {
+  return {
+    ...emptyHeader,
+    headerLeft: () => <BackButton />,
+    headerTitle: i18n.t(`fiatExchangeFlow:${route.params?.isAddFunds ? 'addFunds' : 'cashOut'}`),
+  }
+}
+
+interface Provider {
+  enabled: boolean
+  image: React.ReactNode
+  onSelected: () => void
+}
+
+const FALLBACK_CURRENCY = LocalCurrencyCode.USD
+
+function ProviderOptionsScreen({ route, navigation }: Props) {
+  const [showingExplanation, setShowExplanation] = useState(false)
+  const onDismissExplanation = () => setShowExplanation(false)
+
+  const { t } = useTranslation(Namespaces.fiatExchangeFlow)
+  const account = useSelector(currentAccountSelector)
+  const localCurrency = useSelector(getLocalCurrencyCode)
+  const isAddFunds = route.params?.isAddFunds ?? true
+  const { MOONPAY_DISABLED } = useCountryFeatures()
+
+  useLayoutEffect(() => {
+    const showExplanation = () => setShowExplanation(true)
+
+    navigation.setOptions({
+      headerRightContainerStyle: { paddingRight: 16 },
+      headerRight: () => (
+        <TopBarIconButton
+          icon={<QuestionIcon color={colors.greenUI} />}
+          onPress={showExplanation}
+        />
+      ),
+    })
+  }, [navigation])
```

Comment:
Can we be more specific on when we want this effect to take place? It does a shallow comparison of items in the dependency array so this will run virtually every time.

I'm not sure exactly when you'd like this to run but maybe `navigation.isFocused` works?

Reply:
I added a `console.log` there and the hook is being called only once no matter what I do, which is the intended behaviour since we don't need to run this more than once. I did make the dependency array empty though since that's what we actually want.